src/Logger-HP_3456A/main.py

```diff
@@ -0,0 +1,196 @@
+            "O.C. 2W-Res.": "S1 F4 Z0", #Offset-Compensated 2-Wire Resistance Measurement
+            "O.C. 4W-Res.": "S1 F5 Z0" #Offset-Compensated 4-Wire Resistance Measurement
+        }
+
+        # this dictionary sets the unit of each mode
+        self.mode_units = {
+            "Voltage DC": "V",
+            "Voltage AC": "V",
+            "Voltage AC+DC": "V",
+            "2W-Resistance": "Ohm",
+            "4W-Resistance": "Ohm",
+            "O.C. 2W-Res.": "Ohm",
+            "O.C. 4W-Res.": "Ohm"
+        }
+        #number of displayed digits as dictionary
+        self.resolutions = {
+            "Auto": "R1", 
+            "100 mV or 0.1 kOhm": "R2",
+            "1 V or 1 kOhm": "R3",
+            "10 V or 10 kOhm": "R4",
+            "100 V or 100 kOhm": "R5",
+            "1000 V or 1 MOhm": "R6",
+            "10 MOhm": "R7",
+            "100 MOhm": "R8",
+            "1 GOhm": "R9"
+        }
+
+        # measuring range as dictionary
+        self.ranges = {
+            "6 Digits (Standard)": "6STG",
+            "5 Digits": "5STG",
+            "4 Digits": "4STG",
+            "3 Digits": "3STG"
+        }
+
+        self.nplc_types = {
+            "Medium (1)": "1STI",
+            "Slow (10)": "10STI",
+            "Very Slow (100)": "100STI"
+        }
+
+    def set_GUIparameter(self):
+
+        GUIparameter = {
+            "Mode": list(self.modes.keys()),
+            "Resolution": list(self.resolutions.keys()),
+            "Range": list(self.ranges.keys()),
+            "NPLC": list(self.nplc_types.keys()),
+            "Auto-Zero": ["On", "Off"],
+            "Filter": ["Off", "On"],
+            "Display": ["On", "Off"],
+        }
+
+        return GUIparameter
+
+    def get_GUIparameter(self, parameter={}):
+
+        self.mode = parameter['Mode']
+        self.resolution = parameter['Resolution']
+        self.range = parameter['Range']
+        self.nplc = parameter['NPLC']
+        self.autozero = parameter['Auto-Zero']
+        self.filter = parameter['Filter']
+        self.display = parameter['Display']
+
+        self.port_string = parameter["Port"]
+
+        # here, the variables and units are defined, based on the selection of the user
+        # we have as many variables as channels are selected
+        self.variables = [self.mode]  # we add the channel name to each variable, e.g "Voltage DC"
+
+        self.units = [self.mode_units[self.mode]]
+
+        self.plottype = [True]  # True to plot data
+        self.savetype = [True]  # True to save data
+
+    def initialize(self):
+
+        self.port.write("CL1")  # reset all values
+
+    def deinitialize(self):
+
+        pass
+
+    def configure(self):
+
+        # Mode
+        self.port.write("%s" % self.modes[self.mode])
+
+        # Resolution
+        print(self.resolution)
+        print(self.mode)
```

Comment:
If these debugs/prints are necessary, consider adding a `debug: bool = True` parameter to the GUI

Reply:
In future there might be a general debug option available that tells a driver to be verbose. From this point of view I would rather not add a debug option to the GUI, but just remove the prints to have a clean silent driver and whenever new featues are added, one can add new print statement to check something.

Reply:
Yes, please remove. I like to have the IDs print out in my drivers, but they should not be included in the public drivers and I always forget to remove them.